primers/pcr/pcr.go

```diff
@@ -0,0 +1,138 @@
+	sequence = strings.ToUpper(sequence)
+	forwardPrimer := sequence[0:minimalPrimerLength]
+	for i := 0; primers.MeltingTemp(forwardPrimer) < targetTm; i++ {
+		forwardPrimer = sequence[0 : minimalPrimerLength+i]
+	}
+	reversePrimer := transform.ReverseComplement(sequence[len(sequence)-minimalPrimerLength:])
+	for i := 0; primers.MeltingTemp(reversePrimer) < targetTm; i++ {
+		reversePrimer = transform.ReverseComplement(sequence[len(sequence)-(minimalPrimerLength+i):])
+	}
+
+	// Add overhangs to primer
+	forwardPrimer = forwardOverhang + forwardPrimer
+	reversePrimer = transform.ReverseComplement(reverseOverhang) + reversePrimer
+
+	return forwardPrimer, reversePrimer
+}
+
+func DesignPrimers(sequence string, targetTm float64) (string, string) {
+	return DesignPrimersWithOverhangs(sequence, "", "", targetTm)
+}
+
+func Simulate(sequence string, targetTm float64, circular bool, primerList []string) []string {
+	sequence = strings.ToUpper(sequence)
+	sequenceIndex := suffixarray.New([]byte(sequence))
+
+	primerLength := len(primerList)
+
+	forwardLocations := make(map[int][]int)
+	reverseLocations := make(map[int][]int)
+	minimalPrimers := make([]string, primerLength)
+	for primerIdx, primer := range primerList {
+		var minimalLength int
+		for i := 10; primers.MeltingTemp(primer[len(primer)-i:]) < targetTm; i++ {
+			minimalLength = i
+			if primer[len(primer)-i:] == primer {
+				break
+			}
+		}
+		// Use the minimal binding sites of the primer to find positions in the genome
+		if primer[len(primer)-minimalLength:] != primer {
+			minimalPrimer := primer[len(primer)-minimalLength:]
+			minimalPrimers[primerIdx] = minimalPrimer
+			for _, forwardLocation := range sequenceIndex.Lookup([]byte(minimalPrimer), -1) {
+				forwardLocations[forwardLocation] = append(forwardLocations[forwardLocation], primerIdx)
+			}
+			for _, reverseLocation := range sequenceIndex.Lookup([]byte(transform.ReverseComplement(minimalPrimer)), -1) {
+				reverseLocations[reverseLocation] = append(reverseLocations[reverseLocation], primerIdx)
+			}
+		}
+	}
+
+	var forwardLocationInts []int
+	var reverseLocationInts []int
+	for key := range forwardLocations {
+		forwardLocationInts = append(forwardLocationInts, key)
+	}
+	for key := range reverseLocations {
+		reverseLocationInts = append(reverseLocationInts, key)
+	}
+	sort.Ints(forwardLocationInts)
+	sort.Ints(reverseLocationInts)
+
+	var pcrFragment string
+	var pcrFragments []string
+
+	// Next, iterate through the forwardLocations list
+	for i, forwardLocation := range forwardLocationInts {
+		// First, make sure that this isn't the last element in forwardLocations
+		if i+1 != len(forwardLocationInts) {
+			// If this isn't the last element in forwardLocations, then we can select the first reverseLocation that is less than the next forwardLocation
+			for _, reverseLocation := range reverseLocationInts {
```

Comment:
This block seems to be repeated three times in this loop based on very minimal conditionals. Can this be refactored into a helper function for better code clarity?